web/app/components/input/input.jsx

```diff
@@ -92,16 +93,11 @@ export default class Input extends Component {
         this.setState({ preview: null, text: '' });
       })
       .catch(e => {
-        if (
-          e.response &&
-          e.response.data &&
-          typeof e.response.data.error === 'string' &&
-          e.response.data.error.indexOf("parent comment with reply can't be edited") === 0
-        ) {
-          this.setState({ isErrorShown: true, errorMessage: 'Comment has reply, editing is not possible' });
-          return;
+        let errorMessage = extractErrorMessageFromResponse(e.response);
+        if (errorMessage.indexOf("parent comment with reply can't be edited") === 0) {
```

Comment:
It looks like it's better to ask @umputun to return with error message some error code, because it's better to just have a map `code => text` than rely on text comparison.

Reply:
I agree, it would be much easier to understand what happened if we have error code.
Another option is to have user-friendly error message we've got from the server side.

Reply:
just to make sure: are you asking for an extra code to be added to [this json response](https://github.com/umputun/remark/blob/master/backend/app/rest/httperrors.go#L18) to map it on UI side to smth custom? And the reason for this? I mean why not to show the error text as-is, directly from the `details` field.

Reply:
Correct. If no languages beside the English are supported details should work fine. But seems like we can make it a little bit more UI friendly. Now values for the **details** are something like **"user blocked"** or **"old post, read-only"**. How about at least to make first letter capital and add dot in the end? Or even make it sound like complete sentence (ex. "You can not edit old posts"). Also sometimes the **details** field does not contain real error cause. For instance when user tried to edit comment having reply we've got following error:

```
{
   "details":"can't update comment",
   "error":"parent comment with reply can't be edited, a5bf336f-cb42-4b7d-980d-ed7c21aaebc0"
}
```
In order to provide better explanation for user we have to check the **error** as well.

Reply:
makes sense

Reply:
I have added `code` field to the error response. This is the list of all codes https://github.com/umputun/remark/blob/master/backend/app/rest/httperrors.go#L15

Reply:
I have changed the frontend part so it uses the `code` value.
@umputun I've also changed 1 go file. I believe it was typo, please check.
@igoradamenko Please review the code again.